Some tests in the dot package fail on MacOS

## Describe the bug

Two tests in the `dot` package fail on MacOS: `Test_createSystemService` and `Test_createRuntimeStorage`.

They fail during cleanup, not because any of the assertions aren't met. The error message suggests that it is related to the `unlinkat` syscall (see logs below). The temporary directory where the node DB is stored is not successfully removed after test execution.

This is probably because the process still has a file descriptor open for the file mentioned in the error message. On Linux, `unlinkat` probably removes this file anyway, whereas it does not do this on MacOS.

On MacOS the man page for `unlinkat` mentions `ENOTEMPTY` in the list of possible errors:

```
~$ uname
Darwin
~$ man unlinkat | grep ENOTEMPTY
     [ENOTEMPTY]        The flag parameter has the AT_REMOVEDIR bit set and
```

Whereas on Linux it does not:

```
~$ uname
Linux
~$ man unlinkat | grep ENOTEMPTY
~$
```

## Possible Solution

Make sure no files in the temporary directory used for the test are still open.

## To Reproduce
Steps to reproduce the behaviour:

On MacOS, execute `go test -run "^\QTest_createRuntimeStorage\E$" ./dot`.

## Log output

```
2024-10-29T15:33:33+07:00 INFO     🕸️  initialising node with name Gossamer, id gssmr, base path /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001 and chain-spec /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/002/genesis.json...        pkg=dot
2024-10-29T15:33:33+07:00 INFO     🔨 finalised block #0 (0x276bfa91f70859348285599321ea96afd3ae681f0be47d36196bac8075ea32e8), round 0, set id 0        pkg=state
2024-10-29T15:33:33+07:00 INFO     block state hash genesis hash: 0x276bfa91f70859348285599321ea96afd3ae681f0be47d36196bac8075ea32e8    pkg=state
2024/10/29 15:33:33 [JOB 1] WAL file /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db/000002.log with log number 000002 stopped reading at offset: 0; replayed 0 keys in 0 batches
2024/10/29 15:33:33 [JOB 1] WAL file /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db/000004.log with log number 000004 stopped reading at offset: 0; replayed 0 keys in 0 batches
2024/10/29 15:33:33 [JOB 1] WAL file /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db/000005.log with log number 000005 stopped reading at offset: 0; replayed 0 keys in 0 batches
2024/10/29 15:33:33 [JOB 1] WAL file /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db/000006.log with log number 000006 stopped reading at offset: 1159913; replayed 159 keys in 15 batches
2024-10-29T15:33:33+07:00 INFO     node initialised with name Gossamer, id gssmr, base path /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001, chain-spec /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/002/genesis.json, block 0 and genesis hash 0x276bfa91f70859348285599321ea96afd3ae681f0be47d36196bac8075ea32e8     pkg=dot
2024/10/29 15:33:34 [JOB 1] WAL file /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db/000008.log with log number 000008 stopped reading at offset: 43; replayed 1 keys in 1 batches
--- FAIL: Test_createRuntimeStorage (0.97s)
    testing.go:1232: TempDir RemoveAll cleanup: unlinkat /var/folders/lb/g48kcpxs0fbdnw09nr3633xm0000gn/T/Test_createRuntimeStorage1216796990/001/db: directory not empty
FAIL
FAIL    github.com/ChainSafe/gossamer/dot       1.668s
FAIL
```

## Specification

- go version: 1.23.2
- gossamer commit hash: `a78892d99db2d1c780bcdafa9e53a4fd4cb76423`
- operating system: MacOS (darwin/arm64)

--- a/dot/services_test.go
+++ b/dot/services_test.go
@@ -30,6 +30,8 @@ func Test_createRuntimeStorage(t *testing.T) {
 	stateSrvc, err := builder.createStateService(config)
 	require.NoError(t, err)
 
+	require.NoError(t, stateSrvc.Start())
+
 	tests := []struct {
 		name           string
 		service        *state.Service
@@ -51,6 +53,8 @@ func Test_createRuntimeStorage(t *testing.T) {
 			assert.NotNil(t, got.PersistentStorage)
 		})
 	}
+
+	require.NoError(t, stateSrvc.Stop())
 }
 
 func Test_createSystemService(t *testing.T) {
@@ -64,6 +68,8 @@ func Test_createSystemService(t *testing.T) {
 
 	stateSrvc, err := builder.createStateService(config)
 	require.NoError(t, err)
+
+	require.NoError(t, stateSrvc.Start())
 
 	type args struct {
 		cfg     *types.SystemInfo
@@ -96,6 +102,8 @@ func Test_createSystemService(t *testing.T) {
 			}
 		})
 	}
+
+	require.NoError(t, stateSrvc.Stop())
 }
 
 func Test_newInMemoryDB(t *testing.T) {

